libraries/stdlib/src/kotlin/annotations/NativeAnnotations.kt

```diff
@@ -16,3 +16,18 @@ package kotlin.native
 @Retention(AnnotationRetention.BINARY)
 @OptionalExpectation
 public expect annotation class CName(val externName: String = "", val shortName: String = "")
+
+/**
+ * Instructs the Kotlin compiler to use a custom Objective-C and/or Swift name for this class, property, parameter, function or file.
+ */
+@Target(
+    AnnotationTarget.CLASS,
+    AnnotationTarget.PROPERTY,
+    AnnotationTarget.VALUE_PARAMETER,
```

Comment:
The reasoning behind this is to provide an easy way to make the Kotlin API more ObjC/Swift like.
If users could instead define the whole selector, we would need much more error handling and logic to combine this with the compiler logic that generates selectors for the other declarations.

IMO the current naming logic works great most of the time. It's more about the little things like:
```kotlin
@ObjCName("with")
fun withUserId(userId: String): String = userId
fun scanForPeripherals(@ObjCName("withServices") serviceUUIDs: Int, options: String): String
```
and of course to manually resolve conflicts (like in [KT-50767](https://youtrack.jetbrains.com/issue/KT-50767)).

Reply:
So, the main reason is "it is easier to implement safely and consistently"? Works for me.

Generally, current implementation doesn't allow changing the names of
- completion handler
- error out parameter
- extension receiver

I guess, the latter can actually be supported (Kotlin [can](https://kotlinlang.org/docs/annotations.html#annotation-use-site-targets) apply annotation to an extension receiver with `@receiver:ObjCName`). Could you please try to do that?

And the first two are not visible in "modern" Swift -- translated to `throws` and `async`. So, this would kind of leave us with the empty list of unsupported cases in Swift, which is great.

Reply:
> So, the main reason is "it is easier to implement safely and consistently"? Works for me.

Correct.

> I guess, the latter can actually be supported (Kotlin [can](https://kotlinlang.org/docs/annotations.html#annotation-use-site-targets) apply annotation to an extension receiver with `@receiver:ObjCName`). Could you please try to do that?

Sure! Extension receivers are currently [hardcoded](https://github.com/JetBrains/kotlin/blob/934b8c890e8ea3a489e181f76c834e7c427884d2/kotlin-native/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/objcexport/ObjCExportNamer.kt#L468) to not have a name. 
But the annotation can indeed be applied to the receiver, so we should update that to support the annotation as well.